docs/advanced/Working_with_git.rst

```diff
@@ -0,0 +1,151 @@
+==================
+Working with git
+==================
+
++----------------------------------+-----------------------------------------------------+
+| Git Commands                     |                What it does                         |
++==================================+=====================================================+
+| git **clone**  <repo>            | Used to clone the repo ``git clone <repo> | <name>``|
++----------------------------------+-----------------------------------------------------+
+| git **commit**                   |  **Commit** an applied change on the given branch   |
++----------------------------------+-----------------------------------------------------+
+| git **remote**                   | `track <http://bit.ly/_fo_rk>`_ remote branch       |
```

Comment:
I think [google url shortner](https://goo.gl/ ) can be used ,they even provide an 
an [api](https://developers.google.com/url-shortener/v1/getting_started) for that...?? If we handle the api key to the maintainers or use the **public shortner** ...will it be fine...??
I saw a similar pull request  #1494 using google custom search...I hope we have control over that ...??